src/api/process-manager.ts

```diff
@@ -1,129 +1,167 @@
-import * as uuid from 'uuid';
 import { getRepository } from './db/repository';
-import { startBuildProcess, processes, Process, exitProcess } from './process';
-import { Observable, BehaviorSubject } from 'rxjs';
+import { startBuildJob, Job, startDockerImageSetupJob } from './process';
+import { Observable, Subject } from 'rxjs';
 import { insertBuild, updateBuild, getBuild } from './db/build';
+import { getRepositoryDetails, generateCommands } from './config';
+import { killContainer } from './docker';
 
-export function startBuild(repositoryId: number): Promise<Process> {
-  return new Promise((resolve, reject) => {
-    getRepository(repositoryId).then(data => {
-      const uuid = generateId();
-
-      insertBuild({ uuid: uuid, repositories_id: repositoryId }).then(insertedBuild => {
-        let proc = startBuildProcess(uuid, repositoryId);
-        proc.status = 'starting';
-        const splitted = data.url.split('/');
-        const name = splitted[splitted.length - 1].replace(/\.git/, '');
-
-        let cmds = [
-          '/etc/init.d/xvfb start',
-          'export DISPLAY=:99',
-          'export CHROME_BIN=chromium-browser',
-          `git clone --depth 50 -q ${data.url} ${name}`,
-          `cd ${name}`,
-          'yarn',
-          'yarn test',
-          'exit $?'
-        ].forEach(command => {
-          proc.pty.next({ action: 'command', message: command });
-        });
+export interface BuildMessage {
+  type: string;
+  jobMessage: JobMessage;
+}
 
-        processes.push(proc);
+export interface JobMessage {
+  build_id?: number;
+  job_id?: number;
+  image_name?: string;
+  type: string;
+  data: string | number;
+}
 
-        proc.pty.subscribe(data => {
-          if (proc.status !== 'running') {
-            proc.status = 'running';
-          }
+export interface JobProcess {
+  build_id?: number;
+  job_id?: number;
+  image_name?: string;
+  job: Subject<JobMessage> | Observable<JobMessage> | any;
+}
 
-          if (data.type === 'data') {
-            proc.log.push(data.data);
-          } else if (data.type === 'exit') {
-            updateBuild({
-              id: insertedBuild.toJSON().id,
-              status: data.status,
-              log: proc.log.join('\n')
-            })
-            .then(() => {
-              proc.exitStatus = data.data;
-              proc.status = proc.exitStatus === 0 ? 'success' : 'errored';
-
-              exitProcess(uuid);
-            });
-          }
-        });
+export let jobProcesses: JobProcess[] = [];
 
-        resolve(proc);
-      });
+export function startBuild(): Promise<null> {
+  let gitUrl = 'https://github.com/jkuri/bterm.git';
```

Comment:
Why don't you put this into the config? Also, isn't our gitURL `https://github.com/bleenco/bterm.git`?